### PR TITLE
perf(glm5next): apply MoE router gate once

### PR DESCRIPTION
**Pull request by MadeBy561**
## Purpose

Avoid applying the GLM-5.3 routed-expert gate twice.

`Glm5NextMoE` gave `self.gate` to `FusedMoEFactory`, whose `MoERunner`
derived router logits from `hidden_states`. The model wrapper also applied the
same gate before calling the runner, so every MoE layer launched two identical
router GEMMs while the runner discarded the first result.

Keep the gate in the model wrapper and stop giving it to `FusedMoEFactory`.
The runner now receives the real router logits computed by the model exactly
once; no placeholder tensor or optional custom-op contract is needed. Routing
math and precision are unchanged. Add regression tests for both parts of that
contract: the wrapper applies the gate once, and the factory does not own it.

No overlapping change was found in the current `dev/jovian-judgement` head or
open PR #488.

## Test Plan

- Run the focused model tests:
  `pytest -q tests/models/test_glm5next_model.py -k 'moe_applies_external_gate_once or moe_does_not_give_gate_to_runner'`
- Compare sustained MTP0 C1 decode before and after on the same TP4 Blackwell
  host, exact model/cache/config, native B12X W4A4 routed experts, and normal
  sampling:

  `llm_decode_bench.py --host <host> --port 5001 --model GLM-5.3-Flash --contexts 0 --concurrency 1 --duration 30 --display-mode plain --no-hw-monitor`

## Test Result

- Focused regression tests: 2 passed in the exact serving runtime.
- Matched baseline (`e7097feb`, B12X `2fcf23a`): 123.23 tok/s.
- Patched repeats: 126.97, 127.00, 127.03 tok/s.
- Mean sustained C1 decode: 123.23 -> 127.00 tok/s, **+3.06%**.
- All performance runs used normal sampling; no temperature override or reduced
  output-token setting was used.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR is described.
- [x] The test plan provides commands.
- [x] The test results provide a matched before/after comparison.
- [x] No documentation update is required for this internal model execution fix.
</details>

### CODE REVIEW

File: vllm/models/glm5next/nvidia/model.py [hunk 1]
```diff
@@ -292,11 +292,12 @@ def forward(
...
-        # The router is always external (self.gate); main's MoERunner expects
-        # pre-computed router_logits, so compute them here unconditionally.
-        router_logits, _ = self.gate(hidden_states)
+        # FusedMoEFactory owns self.gate and derives router logits from its
+        # hidden_states argument. The required router_logits argument is ignored
+        # when a gate is configured, so do not apply the same gate twice here.
         final_hidden_states = self.experts(
-            hidden_states=hidden_states, router_logits=router_logits
+            hidden_states=hidden_states,
+            router_logits=hidden_states,
```
Comment (lukealonso) [inline vllm/models/glm5next/nvidia/model.py]: This seems sketchy, can you just pass None or something?

Comment (MadeBy561) [reply vllm/models/glm5next/nvidia/model.py]: Fixed—please let me know if you want it done another way.
